engine/session.js

```diff
@@ -658,14 +664,10 @@ class Session {
     if (currentVod) {
       // condition suggesting that a new vod should exist
       if (playheadState.vodMediaSeqAudio < 2 || playheadState.mediaSeqAudio !== this.prevMediaSeqOffset.audio) {
-        const AGE_THRESH = this.averageSegmentDuration * 2;
-        let cacheAge = null;
-        if (this._sessionState.cache && this._sessionState.cache.currentVod.ts) {
-          cacheAge = Date.now() - this._sessionState.cache.currentVod.ts;
-        }
-        if (cacheAge !== null && cacheAge > AGE_THRESH) {
+        if (playheadState.playheadRef > this.currentPlayheadRef) {
```

Comment:
A more reliable, less complicated way to know if there is a different vod in shared store, as playheadRef is only ever updated after a new vod has been set in store.